Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerConfigurationCocoa.mm

```diff
@@ -105,6 +105,25 @@
     return m_declarativeNetRequestStoreDirectory;
 }
 
+String WebExtensionControllerConfiguration::registeredContentScriptsDirectory()
+{
+    if (!m_registeredContentScriptsDirectory.isEmpty())
+        return m_registeredContentScriptsDirectory;
+
+    if (!storageIsPersistent()) {
+        m_registeredContentScriptsDirectory = FileSystem::createTemporaryDirectory(@"RegisteredContentScripts");
```

Comment:
We should use in-memory DBs for this case, so we don't put things on disk. dNR needed on-disk storage to work, so that is why it is an exception.